### ISSUE TITLE
ERROR: for ddev-router  Cannot start service ddev-router:

### ISSUE THREAD

**Issue by stevekessler** [drud/ddev#3101]
https://gist.github.com/stevekessler/de6164817a3236adccbbe341a71a083e

**Describe the Problem**

When I try and start a DDEV instance I get the following errors.  

ERROR: for ddev-router  Cannot start service ddev-router: Ports are not available: listen tcp 127.0.0.1:80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

ERROR: for ddev-router  Cannot start service ddev-router: Ports are not available: listen tcp 127.0.0.1:80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
Encountered errors while bringing up the project.
Failed to start d8taxonomy: failed to start ddev-router: Failed to run docker-compose [-f /home/steve/.ddev/.router-compose-full.yaml -p ddev-router up -d], err='exit status 1', stdout='', stderr='Creating ddev-router ...
Creating ddev-router ... error

ERROR: for ddev-router  Cannot start service ddev-router: Ports are not available: listen tcp 127.0.0.1:80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.

ERROR: for ddev-router  Cannot start service ddev-router: Ports are not available: listen tcp 127.0.0.1:80: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
Encountered errors while bringing up the project.'

**To Reproduce**

To reproduce I run `ddev start` in a project folder.
I do not think it is project or project type specific since the same problem happens with the diagnostics script (see above).

**Version and configuration information (please complete the following information):**

This is my config.yaml file:

```
name: d8taxonomy
type: drupal8
docroot: web
php_version: "7.4"
webserver_type: nginx-fpm
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: []
mariadb_version: "10.3"
mysql_version: ""
use_dns_when_possible: true
composer_version: ""
web_environment: []
```

**Additional context**
I think that ddev worked for me on this computer but it has been almost a year since I would have ran it so I do not remember because I have a computer I was using it on at that time.  I have tried both a reused configuration and the above file very simple configuration and run into the same problem.  I run into the same problem even if I change the ports that are used.  

I appreciate any I help.  

Thanks,
Steve

**Comment by rfay** [comment]
Hi Steve - Thanks for the diagnostic. It looks like you're using WSL2, and it looks like perhaps you have a docker problem or else something else competing for port 80

* Please show the output of `wsl -l -v` in PowerShell or git-bash (on the Windows side, not inside wsl2)
* What distro are you using? Ubuntu 20.04?
* Have you already tried restarting docker, restarting WSL2, and rebooting the computer?
* The official way to use docker with ddev is to use Docker Desktop... is it possible you also installed docker inside the container?

**Comment by rfay** [comment]
Other things to remember:
* Take a look at the [troubleshooting for occupied ports](https://ddev.readthedocs.io/en/stable/users/troubleshooting/#webserver-ports-are-already-occupied-by-another-webserver)
* You may want to just change the ports you're using

**Comment by rfay** [comment]
Closing, no response from OP. I think the suggestions here will do you well enough. I'd always start with `ddev poweroff` and perhaps restarting docker.

**Comment by stevekessler** [comment]
Hi @rfay,
Thank you for your help. I am sorry it has taken so long for me to reply to this.   I had something thrown at me that became higher priority for me. ::sigh::

For anyone looking back at this I am not sure what "fixed" the problem. I did all the steps above and worked through the Docker port conflicts documentation referenced above.  

I ended up finding that 8081 and 8443 worked for me.  I know idea why other ports would not and custom ports did not work when I opened the ticket. 

Thanks again for the help.  

-Steve
